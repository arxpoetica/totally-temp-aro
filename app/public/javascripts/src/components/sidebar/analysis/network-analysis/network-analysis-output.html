<div>
  <div style="display: inline-block;width: 100%">
    <em ng-if="!$ctrl.showGraph">Output is too large to display at this width</em>
    <div ng-if="!$ctrl.showGraph">
      <button class="pull-left btn btn-primary" ng-click="$ctrl.showOutput()">Expand Results</button>
    </div>
    <div>
      <a class="pull-right btn btn-default btn-sm" ng-href="{{$ctrl.downloadLink}}">
        <span style="color:#4d99e5" class="fa fa-download"></span>
      </a>
    </div>
  </div>
  <!-- Its important to have overflow-x: hidden for the following div, otherwise the scrollbars mess
       up the element width calculation, and the $ctrl.showGraph keeps oscillating once our component
       width becomes exactly equal to the width threshold at which we collapse -->
  <div ng-if="$ctrl.showGraph" style="overflow-x: hidden">
    <network-analysis-output-content></network-analysis-output-content>
  </div>
</div>
