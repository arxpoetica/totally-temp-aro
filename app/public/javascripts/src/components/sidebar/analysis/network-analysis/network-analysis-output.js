class NetworkAnalysisOutputController {
  constructor ($element, $timeout, state) {
    this.$element = $element
    this.$timeout = $timeout
    this.showGraph = false

    this.showOutput = () => {
      state.showNetworkAnalysisOutput = true
    }

    state.plan
      .subscribe((plan) => {
        this.downloadLink = `/reports/network_analysis/download/${plan.id}/4`
      })
  }

  $doCheck () {
    // Show the graph only if the element width is large enough
    var oldShowGraph = this.showGraph

    // A convoluted implementation for a reason - When the component is shown for the first time,
    // this.$element[0].offsetWidth isn't set to the width of the container. setTimeout() takes
    // care of that. IF the showGraph value has changed, then we want to do a $timeout() which
    // calls $scope.$apply(). If we wrap everything in a $timeout(), the digest cycle will keep
    // getting called all the time.
    setTimeout(() => {
      var collapseThresholdInPixels = 300
      this.showGraph = this.$element[0].offsetWidth > collapseThresholdInPixels
      if (oldShowGraph !== this.showGraph) {
        this.$timeout()
      }
    })
  }

  $onInit () {
    // We must apply display:block on $element[0] in order for its size to be reported correctly in $doCheck
    this.$element[0].style.display = 'block'
  }
}

NetworkAnalysisOutputController.$inject = ['$element', '$timeout', 'state']

let networkAnalysisOutput = {
  templateUrl: '/components/sidebar/analysis/network-analysis/network-analysis-output.html',
  bindings: {},
  controller: NetworkAnalysisOutputController
}

export default networkAnalysisOutput
