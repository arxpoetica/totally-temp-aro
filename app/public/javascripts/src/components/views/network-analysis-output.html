<div>
  <p class="pull-right">
    <button class="btn btn-primary" ng-click="$ctrl.showOutput()">Optimization Analysis</button>
  </p>
</div>
