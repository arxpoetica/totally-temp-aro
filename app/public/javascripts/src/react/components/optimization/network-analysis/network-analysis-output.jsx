import React, { Component } from 'react'
import { PropTypes } from 'prop-types'
import Chart from 'chart.js'
import reduxStore from '../../../../redux-store'
import wrapComponentWithProvider from '../../../common/provider-wrapped-component'
import ReportsDownloadModal from '../reports/reports-download-modal.jsx'
import NetworkAnalysisActions from './network-analysis-actions'
import ReportActions from '../reports/reports-actions'
import PlanStates from '../../plan/plan-states'
import deepmerge from 'deepmerge'


export class NetworkAnalysisOutput extends Component {
  constructor (props) {
    super(props)
    this.state = {
      selectedCapitalType: null,
      selectedChartType: null
    }
    this.props.loadReport(this.props.planId)
    this.chartRef = React.createRef()
  }

  componentDidUpdate() {
    if (this.props.chartReportDefinition && !this.state.selectedChartType && !this.state.selectedCapitalType) {
      this.setState({
        selectedChartType: this.props.chartReportDefinition.uiDefinition.chartType[0].chartDefinition.name,
        selectedCapitalType: this.props.chartReportDefinition.uiDefinition.capitalType[0].dataModifiers.labelProperty
      })
    }
  }

  render () {
    const hasChartData = Boolean(this.props.chartReportDefinition && this.props.chartReport && this.props.chartReport.length > 0)
    if (hasChartData) {
      // Why setTimeout()? We need the chart to be rendered with the right display style, THEN we create the chart.
      if (this.state.selectedCapitalType && this.state.selectedChartType) {
        this.updateChartDefinition()
        setTimeout(() => this.updateChart(), 0)
      }
    } else {
      // Someone may have clicked the 'Modify' button to re-run analysis. Clear old chart (if any)
      if (this.chart) {
        this.chart.destroy()
        this.chart = null
      }
    }
    return <div>

      <div className="row p-3" style={{ display: hasChartData ? 'flex' : 'none' }}>
        <div className="col-md-4">
          <label style={{ lineHeight: '36px' }}>Capital Type</label>
        </div>
        <div className="col-md-8">
          <select
            className="form-control"
            value={this.state.selectedCapitalType ? this.state.selectedCapitalType.name : ''}
            onChange={event => this.setState({ selectedCapitalType: event.target.value })}
          >
            {this.props.chartReportDefinition
              ? this.props.chartReportDefinition.uiDefinition.capitalType.map(chart => (
                <option key={chart.chartDefinition.displayName} value={chart.dataModifiers.labelProperty}>
                  {chart.chartDefinition.displayName}
                </option>
              ))
              : null}
          </select>
        </div>
      </div>

      <div className="row pl-3 pr-3 pb-3" style={{ display: hasChartData ? 'flex' : 'none' }}>
        <div className="col-md-4">
          <label style={{ lineHeight: '36px' }}>Chart Type</label>
        </div>
        <div className="col-md-8">
          <select
            className="form-control"
            value={this.state.selectedChartType ? this.state.selectedChartType.name : ''}
            onChange={event => this.setState({ selectedChartType: event.target.value })}
          >
            {this.props.chartReportDefinition
              ? this.props.chartReportDefinition.uiDefinition.chartType.map(chart => (
                <option key={chart.chartDefinition.name} value={chart.chartDefinition.name}>{chart.chartDefinition.displayName}</option>
              ))
              : null}
          </select>
        </div>
      </div>

      <canvas ref={this.chartRef} style={{ display: hasChartData ? 'block' : 'none' }} />

      <div className='alert alert-warning mt-3' style={{ display: hasChartData ? 'none' : 'block' }}>
        Network analysis data not available. Run a new network analysis to see results.
      </div>
      <button className='btn btn-primary pull-left' onClick={() => this.props.showOrHideReportModal(true)}>Reports</button>
      <ReportsDownloadModal reportTypes={['NETWORK_ANALYSIS']} />
    </div>
  }

  componentWillReceiveProps (nextProps) {
    // If the plan state changes (e.g. when optimization is finished), either clear the old report or load the new one
    if (this.props.activePlanState !== nextProps.activePlanState &&
        nextProps.activePlanState === PlanStates.START_STATE) {
      this.props.clearOutput() // not sure this belongs here
    }
  }

  updateChartDefinition () {
    if (!this.props.chartReportDefinition) {
      // This can happen when updateChart() is called from a setTimeout(),
      // and the properties change in the meantime
      return
    }
    var selectedUiDefinition = null
    if (!this.state.selectedCapitalType || !this.state.selectedChartType) {
      selectedUiDefinition = deepmerge(
        this.props.chartReportDefinition.uiDefinition.chartType[0],
        this.props.chartReportDefinition.uiDefinition.capitalType[0]
      )
    } else {
      const selectedChartType = this.props.chartReportDefinition.uiDefinition.chartType.find(item => {
        return item.chartDefinition.name === this.state.selectedChartType
      })
      const selectedCapitalType = this.props.chartReportDefinition.uiDefinition.capitalType.find(item => {
        return item.dataModifiers.labelProperty === this.state.selectedCapitalType
      })
      selectedUiDefinition = deepmerge(selectedChartType, selectedCapitalType)
    }
    const { chartDefinition, dataModifiers } = JSON.parse(JSON.stringify(selectedUiDefinition))
    this.chartDefinition = this.buildChartDefinition(chartDefinition, dataModifiers, this.props.chartReport)
  }

  updateChart () {
    if (this.chart) {
      this.chart.destroy()
      this.chart = null
    }
    var ctx = this.chartRef.current.getContext('2d')
    this.chart = new Chart(ctx, this.chartDefinition)
  }

  buildChartDefinition (rawChartDefinition, dataModifiers, chartData) {
    const { name } = rawChartDefinition

    // BIG HONKIN' NOTE:
    // This filter code was made at the behest of Sir Adam W. Musial of New York fame.
    // Essentially upward outlier IIR(-only) data above the 200% threshold is "not needed"
    // for presentation. THIS CHANGE SHOULD ARGUABLY EXIST AT THE SERVICE LAYER.
    // But since the user can still download ALL data changes, this is a
    // "presentational bandaid" -- only for the sake of filtering out data that
    // end users will need. If the request is ever made to fix this again, we
    // should discuss it at a service level, and not a presentation level.
    // Sincerely, Robert ⚔️ Excalibur ⚔️ Hall
    const filteredData = name.match(/^irr/)
      ? chartData.filter(datum => datum[name] < 2)
      : chartData

    // next, sort the report data
    const sortedData = filteredData.sort((a, b) => {
      const multiplier = (dataModifiers.sortOrder === 'ascending') ? 1.0 : -1.0
      return (a[dataModifiers.sortBy] - b[dataModifiers.sortBy]) * multiplier
    })
    this.populateSeriesValues(sortedData, rawChartDefinition, dataModifiers)
    this.populateAxesOptions(sortedData, rawChartDefinition, dataModifiers)
    this.populateTooltipOptions(rawChartDefinition, dataModifiers)
    return rawChartDefinition
  }

  // Populate a raw chart definition with series values
  populateSeriesValues (sortedData, rawChartDefinition, dataModifiers) {
    const xAxisValues = sortedData.map(item => item[dataModifiers.labelProperty])
    const tickFormat = dataModifiers[rawChartDefinition.name].tickFormat

    // Populate all datasets
    rawChartDefinition.data.datasets.forEach(dataset => {
      dataset.data = sortedData.map((item, index) => {
        // For scatter charts, return an object of the form { x: <value>, y: <value> }, otherwise just return the values
        const y = item[dataset.propertyName] * tickFormat.multiplier
        return (rawChartDefinition.type === 'scatter')
          ? { x: xAxisValues[index], y: y }
          : y
      })
    })
  }

  // Populate the options object to display axes
  populateAxesOptions (sortedData, rawChartDefinition, dataModifiers) {
    const xAxisValues = sortedData.map(item => item[dataModifiers.labelProperty])
    const tickFormat = dataModifiers[rawChartDefinition.name].tickFormat

    // Apply formatting on all y axes
    rawChartDefinition.options.scales.yAxes.forEach(yAxis => {
      yAxis.ticks.callback = (value, index, values) => this.formatAxisValue(value, values,
        tickFormat.prefix || '', tickFormat.suffix || '', tickFormat.precision || 1)
    })

    // Apply formatting on x axis
    const xTickFormat = dataModifiers[dataModifiers.labelProperty].tickFormat
    if (rawChartDefinition.type !== 'scatter') {
      // For scatter charts, we need to set the "data.labels" array
      rawChartDefinition.data.labels = xAxisValues.map(item => this.formatAxisValue(item, xAxisValues, xTickFormat.prefix || '',
        xTickFormat.suffix || '', xTickFormat.precision || 1))
    } else {
      // For all other charts, set the "options.scales.xAxes"
      rawChartDefinition.options.scales.xAxes.forEach(xAxis => {
        xAxis.ticks.userCallback = (value, index, values) => this.formatAxisValue(value, values,
          xTickFormat.prefix || '', xTickFormat.suffix || '', xTickFormat.precision || 1)
      })
    }
  }

  // Populate the options for showing tooltips when a user hovers over a data point
  populateTooltipOptions (rawChartDefinition, dataModifiers) {
    rawChartDefinition.options.tooltips.callbacks.label = (tooltipItem, data) => {
      var allXValues = []
      var allYValues = []
      data.datasets.forEach(dataset => { allXValues = allXValues.concat(dataset.data.map((item, index) => item.x || index)) })
      data.datasets.forEach(dataset => { allYValues = allYValues.concat(dataset.data.map(item => item.y || item)) })
      const xTickFormat = dataModifiers[dataModifiers.labelProperty].tickFormat
      const yTickFormat = dataModifiers[rawChartDefinition.name].tickFormat
      const labelLine1 = `${dataModifiers.labelProperty}: ${this.formatAxisValue(+tooltipItem.label, allXValues, xTickFormat.prefix || '', xTickFormat.suffix || '', xTickFormat.precision || 1)}`
      const labelLine2 = `${rawChartDefinition.displayName}: ${this.formatAxisValue(+tooltipItem.value, allYValues, yTickFormat.prefix || '', yTickFormat.suffix || '', yTickFormat.precision || 1)}`
      return [labelLine1, labelLine2]
    }
  }

  formatAxisValue (value, allValues, tickPrefix, tickSuffix, precision) {
    // This function will format the Y-axis tick values so that we show '100 K' instead of '100000'
    // (and will do the same for millions/billions). We can also specify a tick prefix like '$'
    const maxValue = Math.max.apply(Math, allValues) // Inefficient to do this every time, but 'values' length will be small
    const thresholds = [
      { zeroes: 9, suffix: 'B' }, // Billions
      { zeroes: 6, suffix: 'M' }, // Millions
      { zeroes: 3, suffix: 'K' } // Thousands
    ]
    const threshold = thresholds.filter(item => maxValue >= Math.pow(10, item.zeroes))[0]
    // Two spaces in front of the return value - For some reason values with yMax = 900,000 were getting chopped off on the graph
    // without these two spaces.
    if (threshold) {
      return `  ${tickPrefix}${(value / Math.pow(10, threshold.zeroes)).toFixed(precision)} ${threshold.suffix}${tickSuffix}`
    } else {
      return `  ${tickPrefix}${value.toFixed(precision)}${tickSuffix}` // For values less than 1000
    }
  }

  componentWillUnmount () {
    this.props.clearOutput()
  }
}

NetworkAnalysisOutput.propTypes = {
  isTesting: PropTypes.bool,
  planId: PropTypes.number,
  chartReportMetaData: PropTypes.object,
  chartReportDefinition: PropTypes.object,
  chartReport: PropTypes.array
}

const mapStateToProps = (state) => ({
  planId: state.plan.activePlan.id,
  activePlanState: state.plan.activePlan.planState,
  chartReportMetaData: state.optimization.networkAnalysis.chartReportMetaData,
  chartReportDefinition: state.optimization.networkAnalysis.chartReportDefinition,
  chartReport: state.optimization.networkAnalysis.chartReport
})

const mapDispatchToProps = dispatch => ({
  loadReport: planId => dispatch(NetworkAnalysisActions.loadReport(planId)),
  clearOutput: () => dispatch(ReportActions.clearOutput()),
  showOrHideReportModal: showReportModal => dispatch(ReportActions.showOrHideReportModal(showReportModal))
})

const NetworkAnalysisOutputComponent = wrapComponentWithProvider(reduxStore, NetworkAnalysisOutput, mapStateToProps, mapDispatchToProps)
export default NetworkAnalysisOutputComponent
