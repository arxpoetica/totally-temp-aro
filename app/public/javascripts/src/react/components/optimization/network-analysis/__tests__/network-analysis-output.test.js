/* global test expect jest */
import React from 'react'
import { shallow, mount } from 'enzyme'
import { NetworkAnalysisOutput } from '../network-analysis-output'

const reportMetaData = { id: 10 }
const reportDefinition = {
  uiDefinition: [
    {
      chartDefinition: {
        name: 'irr',
        displayName: 'IRR',
        type: 'scatter',
        data: {
          datasets: [
            {
              label: 'IRR',
              backgroundColor: 'rgb(121, 127, 121)',
              data: [],
              fill: false,
              propertyName: 'irr'
            }
          ]
        },
        options: {
          legend: {
            display: false
          },
          elements: {
            line: {
              fill: false,
              tension: 0
            }
          },
          showLines: true,
          responsive: true,
          title: {
            display: false
          },
          scales: {
            xAxes: [
              {
                display: true,
                labelString: 'CAPEX',
                ticks: {}
              }
            ],
            yAxes: [
              {
                display: true,
                ticks: {
                  beginAtZero: false
                }
              }
            ]
          },
          tooltips: {
            callbacks: {}
          }
        }
      },
      dataModifiers: {
        sortBy: 'index',
        sortOrder: 'ascending',
        labelProperty: 'capex',
        irr: {
          tickFormat: {
            prefix: '',
            suffix: '%',
            multiplier: 100.0
          }
        },
        capex: {
          tickFormat: {
            suffix: '$'
          }
        }
      }
    }
  ]
}
const report = [
  { coverage: 193.0, capex: 25373.196882982724, npv: 299981.76532897586, irr: 0.6926999999999963, index: 1 },
  { coverage: 215.0, capex: 28469.416882982725, npv: 333972.6394671266, irr: 0.6891999999999996, index: 2 },
  { coverage: 11262.0, capex: 1573704.1506588343, npv: 1.2906500787648655E7, irr: 0.5504000000000019, index: 5 },
  { coverage: 13812.0, capex: 1950232.128700931, npv: 1.5683535295389896E7, irr: 0.5437000000000013, index: 6 },
  { coverage: 1647.0, capex: 221904.55623732723, npv: 2102823.3532786667, irr: 0.6023000000000015, index: 3 },
  { coverage: 3580.0, capex: 482498.3552715056, npv: 4402880.713705363, irr: 0.5882000000000004, index: 4 },
  { coverage: 15168.0, capex: 2158455.9218533617, npv: 1.7140295621583264E7, irr: 0.5395000000000001, index: 7 },
  { coverage: 15294.0, capex: 2178230.5915344288, npv: 1.727392685564108E7, irr: 0.5390000000000001, index: 8 }
]

// -----------------------------------------------------------------------------
test('When report and report definition are empty', () => {
  const mockLoadReport = jest.fn()
  const component = shallow(
    <NetworkAnalysisOutput planId={100}
      loadReport={mockLoadReport}
    />
  )
  expect(mockLoadReport).toHaveBeenCalledWith(100)
  expect(component).toMatchSnapshot()
})

// -----------------------------------------------------------------------------
test('When report is empty', () => {
  const component = shallow(
    <NetworkAnalysisOutput planId={100}
      loadReport={() => {}}
      reportDefinition={reportDefinition}
      reportMetaData={reportMetaData}
    />
  )
  expect(component).toMatchSnapshot()
})

// -----------------------------------------------------------------------------
test('When report definition is empty', () => {
  const component = shallow(
    <NetworkAnalysisOutput planId={100}
      loadReport={() => {}}
      report={report}
      reportMetaData={reportMetaData}
    />
  )
  expect(component).toMatchSnapshot()
})

// -----------------------------------------------------------------------------
test('When report and report definition are given', () => {
  const component = mount(
    <NetworkAnalysisOutput planId={100}
      isTesting
      loadReport={() => {}}
      report={report}
      reportDefinition={reportDefinition}
      reportMetaData={reportMetaData}
    />
  )
  expect(component).toMatchSnapshot()
})

// -----------------------------------------------------------------------------
test('When component is unmounted', () => {
  const mockLoadReport = jest.fn()
  const mockClearOutput = jest.fn()
  const component = shallow(
    <NetworkAnalysisOutput planId={100}
      loadReport={mockLoadReport}
      clearOutput={mockClearOutput}
    />
  )
  expect(mockLoadReport).toHaveBeenCalledWith(100)
  component.unmount()
  expect(mockClearOutput).toHaveBeenCalled()
})
