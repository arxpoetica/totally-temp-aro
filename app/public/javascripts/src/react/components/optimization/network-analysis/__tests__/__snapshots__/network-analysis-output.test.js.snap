// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`When report and report definition are empty 1`] = `
<div>
  <div
    className="row p-3"
    style={
      Object {
        "display": "none",
      }
    }
  >
    <div
      className="col-md-4"
    >
      <label
        style={
          Object {
            "lineHeight": "36px",
          }
        }
      >
        Chart type
      </label>
    </div>
    <div
      className="col-md-8"
    >
      <select
        className="form-control"
        onChange={[Function]}
        value=""
      />
    </div>
  </div>
  <canvas
    style={
      Object {
        "display": "none",
      }
    }
  />
  <div
    className="alert alert-warning mt-3"
    style={
      Object {
        "display": "block",
      }
    }
  >
    Network analysis data not available. Run a new network analysis to see results.
  </div>
  <pre
    style={
      Object {
        "display": "none",
      }
    }
  />
  <button
    className="btn btn-primary pull-left"
    onClick={[Function]}
  >
    Reports
  </button>
  <Component />
</div>
`;

exports[`When report and report definition are given 1`] = `
<div>
  <div
    className="row p-3"
    style={
      Object {
        "display": "flex",
      }
    }
  >
    <div
      className="col-md-4"
    >
      <label
        style={
          Object {
            "lineHeight": "36px",
          }
        }
      >
        Chart type
      </label>
    </div>
    <div
      className="col-md-8"
    >
      <select
        className="form-control"
        onChange={[Function]}
        value=""
      >
        <option
          key="irr"
          value="irr"
        >
          IRR
        </option>
      </select>
    </div>
  </div>
  <canvas
    style={
      Object {
        "display": "block",
      }
    }
  />
  <div
    className="alert alert-warning mt-3"
    style={
      Object {
        "display": "none",
      }
    }
  >
    Network analysis data not available. Run a new network analysis to see results.
  </div>
  <pre
    style={
      Object {
        "display": "none",
      }
    }
  >
    {
  "name": "irr",
  "displayName": "IRR",
  "type": "scatter",
  "data": {
    "datasets": [
      {
        "label": "IRR",
        "backgroundColor": "rgb(121, 127, 121)",
        "data": [
          {
            "x": 25373.196882982724,
            "y": 69.26999999999963
          },
          {
            "x": 28469.416882982725,
            "y": 68.91999999999996
          },
          {
            "x": 221904.55623732723,
            "y": 60.23000000000015
          },
          {
            "x": 482498.3552715056,
            "y": 58.820000000000036
          },
          {
            "x": 1573704.1506588343,
            "y": 55.04000000000019
          },
          {
            "x": 1950232.128700931,
            "y": 54.37000000000013
          },
          {
            "x": 2158455.9218533617,
            "y": 53.95000000000001
          },
          {
            "x": 2178230.5915344288,
            "y": 53.90000000000001
          }
        ],
        "fill": false,
        "propertyName": "irr"
      }
    ]
  },
  "options": {
    "legend": {
      "display": false
    },
    "elements": {
      "line": {
        "fill": false,
        "tension": 0
      }
    },
    "showLines": true,
    "responsive": true,
    "title": {
      "display": false
    },
    "scales": {
      "xAxes": [
        {
          "display": true,
          "labelString": "CAPEX",
          "ticks": {}
        }
      ],
      "yAxes": [
        {
          "display": true,
          "ticks": {
            "beginAtZero": false
          }
        }
      ]
    },
    "tooltips": {
      "callbacks": {}
    }
  }
}
  </pre>
  <button
    className="btn btn-primary pull-left"
    onClick={[Function]}
  >
    Reports
  </button>
  <Component />
</div>
`;

exports[`When report definition is empty 1`] = `
<div>
  <div
    className="row p-3"
    style={
      Object {
        "display": "none",
      }
    }
  >
    <div
      className="col-md-4"
    >
      <label
        style={
          Object {
            "lineHeight": "36px",
          }
        }
      >
        Chart type
      </label>
    </div>
    <div
      className="col-md-8"
    >
      <select
        className="form-control"
        onChange={[Function]}
        value=""
      />
    </div>
  </div>
  <canvas
    style={
      Object {
        "display": "none",
      }
    }
  />
  <div
    className="alert alert-warning mt-3"
    style={
      Object {
        "display": "block",
      }
    }
  >
    Network analysis data not available. Run a new network analysis to see results.
  </div>
  <pre
    style={
      Object {
        "display": "none",
      }
    }
  />
  <button
    className="btn btn-primary pull-left"
    onClick={[Function]}
  >
    Reports
  </button>
  <Component />
</div>
`;

exports[`When report is empty 1`] = `
<div>
  <div
    className="row p-3"
    style={
      Object {
        "display": "none",
      }
    }
  >
    <div
      className="col-md-4"
    >
      <label
        style={
          Object {
            "lineHeight": "36px",
          }
        }
      >
        Chart type
      </label>
    </div>
    <div
      className="col-md-8"
    >
      <select
        className="form-control"
        onChange={[Function]}
        value=""
      >
        <option
          key="irr"
          value="irr"
        >
          IRR
        </option>
      </select>
    </div>
  </div>
  <canvas
    style={
      Object {
        "display": "none",
      }
    }
  />
  <div
    className="alert alert-warning mt-3"
    style={
      Object {
        "display": "block",
      }
    }
  >
    Network analysis data not available. Run a new network analysis to see results.
  </div>
  <pre
    style={
      Object {
        "display": "none",
      }
    }
  />
  <button
    className="btn btn-primary pull-left"
    onClick={[Function]}
  >
    Reports
  </button>
  <Component />
</div>
`;
